docs/spec_draft.md

```diff
@@ -1267,6 +1268,137 @@ More formally, `result[i0, ..., iR-1]` is equal to:
+</p>
+
+Conceptually, the kth dimension of `inputs[k]` is dilated by implicitly
+placing `base_dilations[k] - 1` holes between each of the entries in that
+dimension.
+
+`padding[k][0]` and `padding[k][1]` specify the amount of
+padding added at the low-end (next to index 0) and the high-end (next to the
+highest index) of dimension `k` respectively. The amount of padding can be
+negative, where the absolute value of negative padding indicates the number
+of elements to remove from the specified dimension.
+
+Formally, for each k $\in$ [0, N),
+`results[k][j0, j1, ..., jR-1] = map({init_values[k]} U {inputs[k][i0, i1,..., iR'-1]} U {init_values}, body)`
+such that
+  * `jd = floor(((padding[d] + id * base_dilations[d]) + jd * (window_dimensions[d] - window_strides[d])) /  window_dimensions[d])`.
+  * `((padding[d] + id * base_dilations[d]) - jd * window_strides[d]) = 0`. This
+     is to skip the holes in dilated window.
+
+The evaluation order of the reduction function is arbitrary and may be
+non-deterministic. Therefore, the reduction function should not be overly
+sensitive to reassociation. See the discussion about associativity in the
+context of [stablehlo.reduce](#stablehloreduce) for more details.
+
+### Inputs
+
+| Name                | Type                                             | Constraints |
+|---------------------|--------------------------------------------------|-------------|
+| `inputs`            | variadic number of tensors of any supported type | (C1), (C3)  |
+| `init_values`       | variadic number of scalars of any supported type | (C1), (C2)  |
+| `window_dimensions` | 1-dimensional tensor constant type `si64`        | (C4)        |
+| `window_strides`    | 1-dimensional tensor constant type `si64`        | (C5)        |
+| `base_dilations`    | 1-dimensional tensor constant type `si64`        | (C6)        |
+| `window_dilations`  | 1-dimensional tensor constant type `si64`        | (C7)        |
+| `padding`           | 2-dimensional tensor constant type `si64`        | (C8)        |
+| `body`              | `function`                                       | (C9)        |
+
+### Outputs
+
+| Name     | Type                                              | Constraints |
+|----------|---------------------------------------------------|-------------|
+| `results` | variadic number of tensors of any supported type | (C1), (C8), (C9) |
+
+### Constraints
+
+  * (C1) size(`inputs`) $=$ size(`init_values`) $=$ size(`results`) $=$ N and
+         N $\ge$ 1
+
+  * (C2) All `init_values` values have to form an identity under reduction
+         `body`. The behavior of the operation is undefined otherwise.
+
+  * (C3) All `inputs` have the same shape.
+
+  * (C4) On `window_dimensions`
+    * size(`window_dimensions`) $=$ rank(`inputs[k]`) for any k $\in$ [0, N).
+    * `window_dimensions[i]` $\gt 0$ for all i $\in$ [0, size(window_dimensions))
+
+  * (C5) On `window_strides` (optional)
```

Comment:
Do you think we should allow `window_strides` to be potentially larger than the shape of `inputs[k]`? I think leaving this constraint out is fine but just wanted to hear your thoughts.

Reply:
Thanks for bringing this up. 
Like `window_dimensions`, the strides for a dimension d can be more than dim(inputs[k], d) in which case the corresponding window will be discarded for reduction.